packages/sil-voting-app/src/routes/dashboard/components/ProposalsList.tsx

```diff
@@ -1,72 +1,40 @@
-    creationDate: new Date('April 3, 2017 08:00:00'),
-    expiryDate: new Date('May 5, 2023 03:00:00'),
-    quorum: 20,
-    threshold: 14,
-    result: { yes: 4, no: 10, abstain: 3 },
-    vote: 'Yes',
-    status: 'Submitted',
-  },
-  {
-    id: 'zxcvzxcv',
-    title: 'title3',
-    author: 'author3',
-    description: 'short description again',
-    creationDate: new Date('January 3, 2010 04:00:00'),
-    expiryDate: new Date('June 5, 2017 03:00:00'),
-    quorum: 30,
-    threshold: 20,
-    result: { yes: 15, no: 4, abstain: 6 },
-    vote: 'Abstain',
-    status: 'Ended',
-  },
-];
+const renderProposals = (storedProposals: ProposalsState): JSX.Element[] => {
+  const proposals = Object.values(storedProposals).map(state => {
+    return state.proposal;
+  });
 
-const ProposalsList = (): JSX.Element => {
-  const proposals = props.map((_, index) => {
+  return proposals.map((_, index) => {
     if (index === 0) {
-      return <React.Fragment key={index}>{Proposal(props[index])}</React.Fragment>;
+      return <React.Fragment key={index}>{Proposal(proposals[index])}</React.Fragment>;
     }
 
     return (
       <React.Fragment key={index}>
         <Hairline />
-        {Proposal(props[index])}
+        {Proposal(proposals[index])}
       </React.Fragment>
     );
   });
+};
+
+const ProposalsList = (): JSX.Element => {
+  const storedProposals = ReactRedux.useSelector((state: RootState) => state.proposals);
+  const renderedProposals = renderProposals(storedProposals);
 
   return (
     <Block id={PROPOSALS_HTML_ID} flexGrow={1}>
-      {proposals}
+      {renderedProposals}
```

Comment:
You can create component to render proposals and call it like `{storedProposals.map((proposal, index) => <RenderedProposal proposal={proposal} index={index} />)}`

Reply:
I can't do `storedProposals.map()` because it's not an array nor is it iterable, so I've done `proposals.map()` where `proposals` would be `const proposals = Object.values(storedProposals).map(state => state.proposal);`